### ISSUE TITLE
ltshipout: `\AtBeginDvi[label]{code}` is not supported

### ISSUE THREAD

**Issue by muzimuzhi** [latex3/latex2e#740]
## Brief outline of the bug

With the introduction of hook management, now (legacy) commands like `\AtBeginDocument` all accept an optional argument representing the hook label.

But `\AtBeginDvi` doesn't, because of how `\__shipout_add_firstpage_material:Nn` is defined. `\AtEndDvi` doesn't have the same issue because it directly expands to `\AddToHook`.

https://github.com/latex3/latex2e/blob/42a8fa1872e68a53b9551031292c6f633a524d4f/base/ltshipout.dtx#L1817-L1825
https://github.com/latex3/latex2e/blob/42a8fa1872e68a53b9551031292c6f633a524d4f/base/ltshipout.dtx#L1184-L1187
https://github.com/latex3/latex2e/blob/42a8fa1872e68a53b9551031292c6f633a524d4f/base/ltshipout.dtx#L1262-L1274

## Minimal example showing the bug

```latex
\RequirePackage{latexbug}
\documentclass{article}

\AtBeginDvi[label]{\relax} % throws error `Missing \begin{document}.`
\AtEndDvi[label]{\relax}

\begin{document}
\end{document}
```

**Comment by muzimuzhi** [comment]
When this is being resolved, I suggest to add a syntax entry.
```diff
--- a/base/ltshipout.dtx
+++ b/base/ltshipout.dtx
@@ -327,6 +327,9 @@
 % \subsection{Legacy \LaTeX{} commands}
 %
 % \begin{function}{\AtBeginDvi,\AtEndDvi}
+%    \begin{syntax}
+%      \cs{AtBeginDvi} \oarg{label} \Arg{code}
+%    \end{syntax}
 %    \cs{AtBeginDvi} is the existing \LaTeXe{} interface to fill the
 %    \hook{shipout/firstpage} hook. This is not really a good name
 %    as it is not just supporting \texttt{.dvi} but also \texttt{.pdf}
```

**Comment by FrankMittelbach** [comment]
Hi @muzimuzhi , it is true that `\AtBeginDvi` doesn't support the label setting and `\AtEndDvi` only does it by mistake.  We have kept those commands with their current name only for backwards compatibility but the suggestion is to *not* use them for new code, but use the `\AddToHook` machinery directly instead. Perhaps we should have made that clearer in the documentation. We made a deliberate exception for `\AtBeginDocument` and `\AtEndDocument` because they are very widely used already and their names are good.

So in summary there aren't any plans to support an extended syntax for these commands and we will instead make the documentation a bit stronger pointing out that these interfaces are truely legacy and should not be used in new code.
